src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```
@@ -1,16 +1,26 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
+        this.countryDao = new CountryDaoImpl(sessionFactory);
+    }
+
     @Override
     public Country add(Country country) {
-        return null;
+        return countryDao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+
+        return countryDao.get(id).orElse(null);
```

Consider handling the case where the `Optional` is empty more explicitly, possibly by throwing an exception or returning a default value, instead of returning `null`. This can help avoid potential `NullPointerExceptions` in the calling code.